posts/gsoc-2020-blog-3.md

```diff
@@ -0,0 +1,164 @@
+
+- ***Factory Method `newIntegrator()`***: Creates and returns a pointer to an `Integrator` instance of type `itype`.
+    - `Integrator* newIntegrator(const std::string& itype)`;
+    - Declared in **Integrator.h**, line 237 (see this on [GitHub](https://github.com/Cantera/cantera/blob/cf1c0816e7d535a1fc385063aebb8b8e93a85233/include/cantera/numerics/Integrator.h#L237))
+    - Implemented in **ODE_integrators.cpp**, line 13 (see this on [GitHub](https://github.com/Cantera/cantera/blob/cf1c0816e7d535a1fc385063aebb8b8e93a85233/src/numerics/ODE_integrators.cpp#L13))
+
+The header files of different `Integrator` implementations are included near the top of **ODE_integrators.cpp** (see this on [GitHub](https://github.com/Cantera/cantera/blob/cf1c0816e7d535a1fc385063aebb8b8e93a85233/src/numerics/ODE_integrators.cpp#L8)). Cantera only includes one type of `Integrator` by default, the SUNDIALS `CVODES` ODE solver, which automatically installs alongside Cantera. Because `CVODES` was written in C, the `CVodesIntegrator` C++ wrapper is used to access the solver:
+
+- ***Class `CVodesIntegrator`***: A C++ wrapper class for `CVODES`. ([Documentation](https://cantera.org/documentation/docs-2.4/doxygen/html/d9/d6b/classCantera_1_1CVodesIntegrator.html))
+    - Declared in **CVodesIntegrator.h**, line 25 (see this on [GitHub](https://github.com/Cantera/cantera/blob/cf1c0816e7d535a1fc385063aebb8b8e93a85233/include/cantera/numerics/CVodesIntegrator.h#L25))
+    - Implemented in **CVodesIntegrator.cpp**, line 79 (see this on [GitHub](https://github.com/Cantera/cantera/blob/cf1c0816e7d535a1fc385063aebb8b8e93a85233/src/numerics/CVodesIntegrator.cpp#L79))
+    - Included in **ODE_integrators.cpp**, line 8 (see this on [GitHub](https://github.com/Cantera/cantera/blob/cf1c0816e7d535a1fc385063aebb8b8e93a85233/src/numerics/ODE_integrators.cpp#L8))
+
+To create an instance of this `CVODES`-type `Integrator`, the factory method can be called with the "CVODE" keyword:
+
+    newIntegrator("CVODE")
+
+This call returns a generic `Integrator` pointer, whose `virtual` functions are overridden by those of a `CVodesIntegrator`. This is exactly how a `ReactorNet` creates its `CVODES`-type `Integrator` object, before storing it locally as `m_integ` for future reference:
+
+**ReactorNet.cpp**, line 18 (see this on [GitHub](https://github.com/Cantera/cantera/blob/cf1c0816e7d535a1fc385063aebb8b8e93a85233/src/zeroD/ReactorNet.cpp#L18))
+
+    m_integ(newIntegrator("CVODE"))
+
+So, what actually happens when you call one of a `ReactorNet`'s time integration functions? Let's follow a call to `ReactorNet::advance()`, like this one to '`sim`', the `ReactorNet` object in Cantera's combustor example:
+
+**combustor.cpp**, line 122 (see this on [GitHub](https://github.com/Cantera/cantera/blob/cf1c0816e7d535a1fc385063aebb8b8e93a85233/samples/cxx/combustor/combustor.cpp#L122))
```

Comment:
It's good to include a real example here. I think it would make sense to show the example as early as possible in the article, and show some more context, including setting up the Reactor and ReactorNet in user-code. It might also help to show the equivalent Python example, since people are more likely to be familiar with that interface than C++.

Reply:
Added outlines of the combustor examples in C++ and Python as the first part of the article, I linked to the code instead of actually including it mostly just to save space (it would be a lot of code)